pkg/imageacquirer/container-image-lib-image-acquirer.go

```diff
@@ -0,0 +1,246 @@
+		if nil != otherErr {
+			return nil, "", fmt.Errorf("invalid source name %s: %v", source, err)
+		}
+	}
+
+	certPath, err := clia.certPath(source)
+	if err != nil {
+		return nil, "", fmt.Errorf("finding certificate path: %v", err)
+	}
+	sourceCtx := &types.SystemContext{
+		DockerCertPath: certPath,
+	}
+
+	if clia.dstPath, err = util.CreateOutputDir(clia.dstPath, "image-inspector-"); err != nil {
+		return nil, "", fmt.Errorf("creating output dir: %v", err)
+	}
+	destRef, err := directory.NewReference(clia.dstPath)
+	if err != nil {
+		return nil, "", fmt.Errorf("invalid destination name %s: %v", clia.dstPath, err)
+	}
+
+	imagePullAuths, err := getAuthConfigs(clia.Auths)
+	if err != nil {
+		return nil, "", fmt.Errorf("getting registry auth config: %v", err)
+	}
+
+	// Try all the possible auths from the config file
+	log.Println("Pulling image")
+	for name, auth := range imagePullAuths.Configs {
+		sourceCtx.DockerAuthConfig = &types.DockerAuthConfig{
+			Username: auth.Username,
+			Password: auth.Password,
+		}
+		reportReader, reportWriter := io.Pipe()
+		// print progress from reportWriter
+		go func() {
+			lineReader := bufio.NewReader(reportReader)
+			buffered := []byte{}
+			last := time.Now()
+			for {
+				b, err := lineReader.ReadByte()
+				if err != nil {
+					return
+				}
+				if b == byte('\n') {
+					log.Printf("%s", buffered)
+					buffered = []byte{}
+				}
+				buffered = append(buffered, b)
+				if b == byte(']') {
+					if time.Since(last) > PULL_LOG_INTERVAL_SEC {
+						log.Printf("%s", buffered)
+						last = time.Now()
+					}
+					buffered = []byte{}
+				}
+			}
+		}()
+		err = copy.Image(policyContext, destRef, srcRef, &copy.Options{
+			RemoveSignatures: false,
+			SignBy:           "",
+			ReportWriter:     reportWriter,
+			SourceCtx:        sourceCtx,
+			DestinationCtx:   nil,
+			ProgressInterval: PULL_LOG_INTERVAL_SEC,
+		})
+		reportWriter.Close()
+		if err == nil {
+			break
+		}
+		log.Printf("Authentication with %s failed: %v", name, err)
+	}
+
+	if err != nil {
+		return nil, "", fmt.Errorf("pulling docker image: %v\n", err)
+	}
+
+	img, err := srcRef.NewImage(sourceCtx)
+	if err != nil {
+		return nil, "", fmt.Errorf(": %v", err)
+	}
+
+	log.Println("Reading image manifest")
+	rawManifest, _, err := img.Manifest()
+	if err != nil {
+		return nil, "", fmt.Errorf("reading image manifest: %v ", err)
+	}
+	imageDigest, err := manifest.Digest(rawManifest)
+	if err != nil {
+		return nil, "", fmt.Errorf("parsing image manifest: %v ", err)
+	}
+
+	inspectInfo, err := img.Inspect()
+	if err != nil {
+		return nil, "", fmt.Errorf("inspecting copied image Manifest: %v ", err)
+	}
+
+	return inspectInfo, imageDigest, nil
+}
+
+// certPath will try to extract the registry name from the image and return
+// "/etc/docker/certs.d/<REGISTRY_NAME>" if this path exists or nil otherwise.
+func (clia *containerLibImageAcquirer) certPath(fullImageName string) (string, error) {
```

Comment:
This should not be necessary; containers/image/docker uses `/etc/docker/certs.d` automatically.